kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package org.motechproject.nms.kilkari.service.impl;
+
+import org.motechproject.nms.kilkari.domain.Subscriber;
+import org.motechproject.nms.kilkari.repository.SubscriberDataService;
+import org.motechproject.nms.kilkari.service.SubscriberService;
+import org.motechproject.nms.kilkari.service.SubscriptionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Implementation of the {@link SubscriptionService} interface.
+ */
+@Service("subscriberService")
+public class SubscriberServiceImpl implements SubscriberService {
+
+    private SubscriberDataService subscriberDataService;
+
+    @Autowired
+    public SubscriberServiceImpl(SubscriberDataService subscriberDataService) {
+        this.subscriberDataService = subscriberDataService;
+    }
+
+    @Override
+    public Subscriber getSubscriber(long callingNumber) {
+        return subscriberDataService.findByCallingNumber(callingNumber);
+    }
+
+    @Override
+    public void add(Subscriber subscriber) {
+        subscriberDataService.create(subscriber);
```

Comment:
It isn't really a new method, I just moved it out of SubscriptionService.  

Although the Subscriber table is unique in the database so that would throw if this was a dupe.  Is that the check you would want to do in code as well?